### DISCUSSION TITLE
Implement support for multiple rooms in the high-level multiplayer API

### DISCUSSION THREAD

**Discussion by michaldev** [godotengine/godot-proposals#9110] [Engine Core]
### Describe the project you are working on

Game with vehicles. Player must have to destroy enemy's vehicle.

### Describe the problem or limitation you are having in your project

Default solution for multiplayer is simple for working with one server and multiple clients (useless for most solutions). I would like to make easy creating system replication for multiple rooms (dynamically creating rooms by players and joining). Entering ip addresses aren't user-friendly. One server per one room isn't server/scalable-friendly. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

-

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

-

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Probably not (with clean, scalable and "not hack" method). 

### Is there a reason why this should be core and not an add-on in the asset library?

This is part of game engine.

**Comment by Calinou** [comment]
I believe this would be covered by [interest management](https://github.com/godotengine/godot-proposals/issues/3904), although it will still require hacks to be implemented on your end.

> One server per one room isn't server/scalable-friendly.

Godot headless processes don't consume too much RAM individually, so this should be feasible. Automatic scaling can be performed using Kubernetes or the like. It's already being done in production :slightly_smiling_face: 

Also, a big upside of using multiple processes is that it will automatically use multiple CPU threads. This is not the case if you most multiple game servers (or rooms) from the same process.

> Describe the feature / enhancement and how it helps to overcome the problem or limitation
>
> - 

If you do not have an idea for the implementation, please use the [Discussions](https://github.com/godotengine/godot-proposals/discussions) tab instead of creating proposal issues.
